src/v2/Apps/Order/Components/BankAccountPicker.tsx

```diff
@@ -0,0 +1,168 @@
+
+interface Props {
+  order: BankAccountPicker_order
+  me: BankAccountPicker_me
+  onSetPaymentSuccess: () => void
+  onSetPaymentError: (error: Error) => void
+}
+
+export const BankAccountPicker: FC<Props> = props => {
+  const getInitialBankAccountSelection = () => {
+    if (props.order.bankAccountId) {
+      return {
+        type: "existing",
+        id: props.order.bankAccountId,
+      }
+    } else {
+      return props.me.bankAccounts?.edges?.length
+        ? {
+            type: "existing",
+            id: props.me.bankAccounts.edges[0]?.node?.internalID!,
+          }
+        : { type: "new" }
+    }
+  }
+
+  const [bankAccountSelection, setBankAccountSelection] = useState(
+    getInitialBankAccountSelection()
+  )
+
+  const {
+    me: { bankAccounts },
+    order,
+    onSetPaymentSuccess,
+    onSetPaymentError,
+  } = props
+
+  const bankAccountsArray = bankAccounts?.edges?.map(e => e?.node)!
+
+  const userHasExistingBankAccounts = bankAccountsArray.length > 0
+
+  const { submitMutation: setPaymentMutation } = useSetPayment()
+
+  const handleContinue = async () => {
+    try {
+      const orderOrError = (
+        await setPaymentMutation({
+          variables: {
+            input: {
+              id: order.internalID,
+              paymentMethod: "US_BANK_ACCOUNT",
+              paymentMethodId: bankAccountSelection.id,
+            },
+          },
+        })
+      ).commerceSetPayment?.orderOrError
+
+      if (orderOrError?.error) {
+        throw orderOrError.error
+      }
+
+      onSetPaymentSuccess()
+    } catch (error) {
+      onSetPaymentError(error)
+    }
+  }
+
+  return (
+    <>
+      {userHasExistingBankAccounts && (
+        <RadioGroup
+          onSelect={val => {
+            if (val === "new") {
+              setBankAccountSelection({ type: "new", id: "" })
+            } else {
+              setBankAccountSelection({ type: "existing", id: val })
+            }
+          }}
+          defaultValue={
+            bankAccountSelection.type === "new"
+              ? "new"
+              : bankAccountSelection.id
+          }
+        >
+          {bankAccountsArray
+            .map(bank => {
+              const { internalID, last4 } = bank!
+
+              return (
+                <BorderedRadio value={internalID} key={internalID}>
+                  <BankDebitDetails last4={last4} />
+                </BorderedRadio>
+              )
+            })
+            .concat([
+              <BorderedRadio
+                data-test="AddNewCard"
+                value="new"
+                key="new"
+                selected={bankAccountSelection.type === "new"}
+              >
+                Add another bank account.
+              </BorderedRadio>,
+            ])}
+        </RadioGroup>
+      )}
+      <Spacer mb={4} />
+      <Collapse
+        open={
+          bankAccountSelection.type === "new" || !userHasExistingBankAccounts
```

Comment:
Perhaps `userHasExistingBankAccounts` is redundant if we set the initial bank account selection properly.